static/app/components/surfaceProvider.tsx

```diff
@@ -0,0 +1,18 @@
+import {createContext, type ReactNode, useContext} from 'react';
+
+export const SurfaceContext = createContext<string>('');
+
+export default function SurfaceProvider({
```

Comment:
Is there a better name we can use here?

Reply:
Do you have any suggestions?

Reply:
`AnalyticsAreaProvider

Reply:
What about AnalyticsSurfaceProvider?

Reply:
Why the word `Surface`? That has far too much of a UI connotation I think here

Reply:
It was suggested by @ryan953 . I think the UI connotation is intended, we want each surface to correspond to a specific component/part of the UI.

Reply:
Isn't `Area` a bit more clear though?

<img width="819" alt="image" src="https://github.com/user-attachments/assets/3f1ba948-7a30-4212-a09b-75e14027bbe0">

Reply:
**Maybe component isn't the right word here, could be a whole page or section of a page..

Reply:
I'm not super attached to surface, but grepping `surface:` it seems we're using it in a few places in replay frontend already. Replays only, probably Ryan and maybe Michelle

Reply:
How about `AnalyticsSectionProvider` then?

Reply:
I agree with Evan here. If this is intended for analytics, then a prefix to indicate that would be nice. We use the `surface` naming in our theme as a proxy for elevation, so I wouldnt duplicate it.

Aside from this, I always ask myself "As someone that is unfamiliar with the codebase, if I see <insert component/> do I have a rough idea about what it does or do I need to check the implementation?". It might sound like a nit, but it is death by a thousand cuts if we end up with too many instances where it is not easy to understand what a component does.

Reply:
Great points! I agree this name is important, and the prefix definitely makes sense. 

Do we prefer AnalyticsAreaProvider or AnalyticsSectionProvider? Personally I like Area since it seems more UI-specific